### PR TITLE
local preview: add instructions for dev urls

### PR DESCRIPTION
**Pull request by bpmct**
add instructions for running dev urls with the local preview.

significant changes:
- runs local preview script to install on port 80 instead of 8080 (necessary for dev urls)
- dev urls is documented as optional as it requires more work and not necessary (ngrok remains an option)
- for dev urls, added a recommendation for nip.io or dnsmasq to add a wildcard subdomain for Dev URLs

teardown guide for dnsmasq is included

### CODE REVIEW

**Comment by bpmct** [comment]
[ch14942]

File: setup/kubernetes/local-preview.md [hunk 1]
```diff
@@ -4,8 +4,8 @@ description: Set up a Coder deployment locally for testing.
 ---
 
 Coder is typically deployed onto Kubernetes cluster, but if you would like to
-set up a lightweight preview deployment, you can do so locally using using
-[Docker][docker-url] and [kind][kind-url].
+set up a lightweight preview deployment, you can do so locally using using [URL]
+and [URL].
```
Comment (coadler) [inline setup/kubernetes/local-preview.md:8]: I'm a bit confused how these `[URL]` links work

Comment (bpmct) [reply setup/kubernetes/local-preview.md:8]: they don't: my linter messed that up. i'll fix

File: setup/kubernetes/local-preview.md [hunk 2]
```diff
@@ -52,15 +52,16 @@ apply):
 
 1. Docker Desktop for Mac **must** use version [2.5.0.1][docker-mac-url]. This
    specific version is required because of a recent downgrade to Linux Kernel
-   4.9 due to a [bug](docker-bug-url).
+   4.9 due to a [bug][docker-bug-url].
```
Comment (coadler) [inline setup/kubernetes/local-preview.md:55]: This seems to be the only one with brackets around the named url.

File: setup/kubernetes/local-preview.md [hunk 3]
```diff
@@ -52,15 +52,16 @@ apply):
...
-Currently, the local preview doesn't support [dev URLs][devurl-url]. Instead,
-you can use tools like [ngrok][ngrok-url] to preview webpages from inside an
-workspace.
+The local preview doesn't support [dev URLs](../workspaces/devurls) by default.
+Instead, you can use tools like [ngrok][ngrok-url] to preview webpages from
+inside a workspace.
 
-We are working on bringing Dev URL support to local previews in later releases.
+To enable local dev URLs, follow
+[the instructions below](#enable-local-dev-urls) after installing Coder.
```
Comment (coadler) [inline setup/kubernetes/local-preview.md:64]: This seems a bit weird to say devurls aren't supported, then giving instructions on how to enable them.

Comment (bpmct) [reply setup/kubernetes/local-preview.md:64]: agreed, just reworded

**Review by khorne3 — APPROVED** [review]

File: setup/kubernetes/local-preview.md [hunk 4]
```diff
@@ -130,7 +128,7 @@ rebinding protection. Here's how:
    sudo apt-get install dnsmasq
    ```
 
-1. Create dnsmasq configuration for the .coder domain
+1. Create a dnsmasq configuration for the `.coder` domain
```
Comment (jawnsy) [inline setup/kubernetes/local-preview.md:131]: One more option is to modify `/etc/hosts` to manually add records pointing to localhost, it's not as seamless as dnsmasq but will work on Windows too https://en.wikipedia.org/wiki/Hosts_(file)#Location_in_the_file_system

File: setup/kubernetes/local-preview.md [hunk 5]
```diff
@@ -99,26 +89,34 @@ Password: yfu...yu2
...
-ingress:
-  host: "127.0.0.1.nip.io"
-devurls:
-  host: "*.127.0.0.1.nip.io"
-```
+1. To use dev URLs, you must have a wildcard subdomain. One option to meet this
+requirement is to use a service such as [nip.io][nip-url] to route domains local
+IP address.
+
+1. [Update Coder](https://coder.com/docs/coder/latest/setup/updating#update-coder)
```
Comment (jawnsy) [inline setup/kubernetes/local-preview.md:105]: I think this should be a relative link

File: setup/kubernetes/local-preview.md (hunk 4 above)
Comment (bpmct) [reply setup/kubernetes/local-preview.md:131]: Would this be for each dev url added?

File: setup/kubernetes/local-preview.md [hunk 6]
```diff
@@ -97,6 +89,90 @@ Password: yfu...yu2
...
+
+Coder allows you to access services you're developing in your workspace via [dev
+URLs](../../workspaces/devurls.md). You can enable dev URLs after you've
+installed Coder.
+
+> If you do not want to enable dev URLs, you can use SSH port forwarding or
+> tools like [ngrok][ngrok-url] to preview webpages from inside you workspace.
+
+1. To use dev URLs, you must have a wildcard subdomain. One option to meet this
+requirement is to use a service such as [nip.io][nip-url] to route domains local
```
Comment (bpmct) [inline setup/kubernetes/local-preview.md:102]:
```suggestion
requirement is to use a service such as [nip.io][nip-url] to route domains to a local
```

File: setup/kubernetes/local-preview.md (hunk 4 above)
Comment (jawnsy) [reply setup/kubernetes/local-preview.md:131]: Yeah, the hosts files don't support wildcards, so a user would have to manually add like

```
abc.coder.localhost 127.0.0.1
def.coder.localhost 127.0.0.1
```

where they configure their Dev URL suffix domain as `coder.localhost`. admittedly this is a pretty bad user experience.

the option with nip is easier to set up but the URLs are uglier :man_shrugging:

**Review by coadler — APPROVED** [review]

File: setup/kubernetes/local-preview.md (hunk 4 above)
Comment (bpmct) [reply setup/kubernetes/local-preview.md:131]: Yeah, would prefer not to mention this for now

**Comment by khorne3** [comment]
@BrunoQuaresma this is the PR that broke the build. is it possible to get increased logging for instances such as this (whatever the problem may be)?

**Comment by BrunoQuaresma** [comment]
@khorne3 from the logs, the error is: 
```
Unknown language: `conf` is not registered
```

**Comment by BrunoQuaresma** [comment]
Probably, related to this:
```
```conf
   # coder.conf
   address=/coder/127.0.0.1
   ```
```

**Pull request merged**